datameta/api/openapi.yaml

```diff
@@ -163,6 +473,166 @@ components:
         - newPassword
         - passwordChangeCredential
 
+    RegRequest:
+      type: object
+      properties:
+        fullname:
+          type: string
+        email:
+          type: string
+        groupId:
+          type: string
+          nullable: True
+        newGroupName:
+          type: string
+          nullable: True
+      required:
+        - fullname
+        - email
+        # either provide
+        # groupId or newGroupName
+      additionalProperties: false
+    
+    MetaDataSet:
+      type: object
+      properties:
+        records:
+          type: object
+          additionalProperties: true
+          # a free-form object,
+          # any property is allowed
+      required:
+        - records
+      additionalProperties: false
+          
+    MetaDataSetResponse:
+      type: object
+      properties:
+        records:
+          type: object
+          additionalProperties: true
+          # a free-form object,
+          # any property is allowed
+        metaDataSetId:
+          type: string
+        submissionId:
+          type: string
+          nullable: true
+        groupId:
+          type: string
+        userId:
+          type: string
+      required:
+        - records
+      additionalProperties: false
+        
+
+    FileAnnouncement:
+      type: object
+      properties:
+        name:
+          type: string
+        checksum:
+          type: string
+        userId:
```

Comment:
Is it necessary to capture the user ID here?

Reply:
No, actually not, I'll remove it

Reply:
Done